traffic_ops/testing/api/v4/cachegroups_test.go

```diff
@@ -32,99 +35,393 @@ import (
 
 func TestCacheGroups(t *testing.T) {
 	WithObjs(t, []TCObj{Types, Parameters, CacheGroups, CDNs, Profiles, Statuses, Divisions, Regions, PhysLocations, Servers, Topologies}, func() {
-		ReadUpdateTestCacheGroupWithNullLatLong(t)
-		GetTestCacheGroupsIMS(t)
-		GetTestCacheGroupsByNameIMS(t)
-		GetTestCacheGroupsByShortNameIMS(t)
-		GetTestCacheGroups(t)
-		GetTestCacheGroupsByName(t)
-		GetTestCacheGroupsByShortName(t)
-		GetTestCacheGroupsByTopology(t)
-		CheckCacheGroupsAuthentication(t)
-		currentTime := time.Now().UTC().Add(-5 * time.Second)
-		time := currentTime.Format(time.RFC1123)
-		var header http.Header
-		header = make(map[string][]string)
-		header.Set(rfc.IfModifiedSince, time)
-		header.Set(rfc.IfUnmodifiedSince, time)
-		UpdateTestCacheGroups(t)
-		UpdateTestCacheGroupsWithHeaders(t, header)
-		GetTestCacheGroupsAfterChangeIMS(t, header)
-		header = make(map[string][]string)
-		etag := rfc.ETag(currentTime)
-		header.Set(rfc.IfMatch, etag)
-		UpdateTestCacheGroupsWithHeaders(t, header)
-		GetTestPaginationSupportCg(t)
-		GetTestCacheGroupsByInvalidId(t)
-		GetTestCacheGroupsByInvalidType(t)
-		GetTestCacheGroupsByType(t)
-		DeleteTestCacheGroupsByInvalidId(t)
-		UpdateCachegroupWithLocks(t)
+
+		methodTests := map[string]map[string]struct {
+			endpointId     func() int
+			clientSession  *client.Session
+			requestParams  map[string]string
+			requestBody    map[string]interface{}
+			requestHeaders http.Header
```

Comment:
Can we collapse the `requestParams` and `requestHeaders` into the `client.RequestOptions` structure? Looks like you're just rebuilding that below.

Reply:
Yeah, I've gone ahead and changed this per your recommendation, initially I had it this way but the lines were long and verbose so I separated them.